File: public/others/builder_cpp/Director.cpp
class $DIRECTOR_CLASSNAME$
{
private:
    $BUILDER_CLASSNAME$ *builder;

public:
    void setBuilder($BUILDER_CLASSNAME$ *builder)
    {
        this->builder = builder;
    }


    void construct()
    {
        //this->builder->ProducePartA();
        //this->builder->ProducePartB();
        //this->builder->ProducePartC();
    }
}

File: public/creational_patterns/builder/Director.java

public class $DIRECTOR_CLASSNAME$ {
	private $BUILDER_CLASSNAME$ builder;

	public $DIRECTOR_CLASSNAME$($BUILDER_CLASSNAME$ builder) {
		this.builder = builder;
	}

	public void changeBuilder($BUILDER_CLASSNAME$ builder) {
		this.builder = builder;
	}

	public void construct() {
		builder.buildStepA();
		builder.buildStepB();
		builder.buildStepZ();
	}
}